pytensor/link/pytorch/dispatch/basic.py

```diff
@@ -0,0 +1,113 @@
+import warnings
+from functools import singledispatch
+
+# import jax
+# import jax.numpy as jnp
+import torch
+import numpy as np
+
+from pytensor.compile.ops import DeepCopyOp, ViewOp
+from pytensor.configdefaults import config
+from pytensor.graph.fg import FunctionGraph
+from pytensor.ifelse import IfElse
+from pytensor.link.utils import fgraph_to_python
+from pytensor.raise_op import Assert, CheckAndRaise
+
+
+# if config.floatX == "float64":
+#     jax.config.update("jax_enable_x64", True)
+# else:
+#     jax.config.update("jax_enable_x64", False)
+
+
+@singledispatch
+def pytorch_typify(data, dtype=None, **kwargs):
+    r"""Convert instances of PyTensor `Type`\s to PyTorch types."""
+    if dtype is None:
+        return data
+    else:
+        return torch.tensor(data, dtype=dtype)
+
+
+@pytorch_typify.register(np.ndarray)
+def pytorch_typify_ndarray(data, dtype=None, **kwargs):
+    if len(data.shape) == 0:
+        return data.item()
+    return torch.tensor(data, dtype=dtype)
+
+
+@singledispatch
+def pytorch_funcify(op, node=None, storage_map=None, **kwargs):
+    """Create a PyTorch compatible function from an PyTensor `Op`."""
+    raise NotImplementedError(f"No PyTorch conversion for the given `Op`: {op}")
+
+
+@pytorch_funcify.register(FunctionGraph)
+def pytorch_funcify_FunctionGraph(
+    fgraph,
+    node=None,
+    fgraph_name="pytorch_funcified_fgraph",
+    **kwargs,
+):
+    return fgraph_to_python(
+        fgraph,
+        pytorch_funcify,
+        type_conversion_fn=pytorch_typify,
+        fgraph_name=fgraph_name,
+        **kwargs,
+    )
+
+
+@pytorch_funcify.register(IfElse)
+def pytorch_funcify_IfElse(op, **kwargs):
+    n_outs = op.n_outs
+
+    def ifelse(cond, *args, n_outs=n_outs):
+        res = torch.where(
+            cond, lambda _: args[:n_outs], lambda _: args[n_outs:], operand=None
+        )
+        return res if n_outs > 1 else res[0]
+
+    return ifelse
+
+
+@pytorch_funcify.register(Assert)
+@pytorch_funcify.register(CheckAndRaise)
+def pytorch_funcify_CheckAndRaise(op, **kwargs):
+
+    def assert_fn(x, *inputs):
+        return x
+
+    return assert_fn
+
+
+def pytorch_safe_copy(x):
+    try:
+        res = x.clone().detach()
```

Comment:
I don't get the detatch. Wouldn't that produce incorrect gradient graphs if someone call `backward` on pytorch code? Maybe just calling `torch.clone(x)` should do the right thing?
I think this also can just be part of the implementation below, can it?